### PR TITLE
System tables rename global_id -> object_id

### PR DESCRIPTION
**Pull request by chaas**
Global id is confusing as it's not clear to the user what it's global to. Rename all uses of `global_id` in system tables to `object_id`, since the identifier refers back to the `id` in the `mz_objects` table. Also add a note in the system table documentation for `mz_objects` about the shared id namespace.

### Motivation
   * This PR refactors existing code. #14407

### Checklist

- [x] This PR has adequate test coverage / QA involvement has been duly considered.
- [ ] This PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way) and therefore is tagged with a `T-protobuf` label.
- [x] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):

This release will change all columns named "global_id" in system tables to "object_id".

### CODE REVIEW

**Comment by jkosh44** [comment]
Is the idea here that the system tables will use "object_id" instead of "global_id", but internally we will continue to use global id? Or are we eventually planning on changing global id to object id internally?

File: doc/user/content/sql/system-catalog.md [hunk 1]
```diff
@@ -342,6 +342,10 @@ Field          | Type       | Meaning
 The `mz_objects` view contains a row for each table, source, view, materialized view, sink,
 index, connection, secret, type, and function in the system.
 
+IDs for all objects represented in `mz_objects` share a namespace -
```
Comment (mjibson) [inline doc/user/content/sql/system-catalog.md:345]: Light suggestion to remove both the hyphen (nee em dash) and e.g.:

> IDs for all objects represented in `mz_objects` share a namespace. If there is a view with ID u1 ...

If you want to keep them, let's at least use a real em dash.

**Review by mjibson — APPROVED** [review]

**Comment by chaas** [comment]
> Is the idea here that the system tables will use "object_id" instead of "global_id", but internally we will continue to use global id? Or are we eventually planning on changing global id to object id internally?

For now at least, it's just for the system tables. For all the existing usages of `global_id` in the system tables, they're all referring to a record in `mz_objects` and would likely be joining against that table, so it's better to be clear about the foreign key relationships.
It's a good question though for future thought around whether we should continue using the term global id internally too 
cc @benesch .

**Review by jkosh44 — APPROVED** [review]
LGTM

**Pull request merged**